### PR TITLE
Make Graph View clickable and link it to Cromwell's timing diagram.

### PR DESCRIPTION
**Pull request by rexwangcc**
Make the `Gragh View` tab link to the Cromwell's timing diagram by doing:
1. Separate the original `environment.cromwell.ts` to `environment.cromwell.dev.ts` and `environment.cromwell.staging.ts`.
2. Add a new field `serverUrl` to `environment.ts`.
3. Change the `tasks.component.ts` to enable it to use the job id and the serverUrl.
4. Change the `.angular-cli.json` to add the two new Cromwell environments.

Now, to build the UI with the Mint's Cromwell, we need to use:
- For dev Cromwell: `docker-compose -f cromwell-compose-dev.yml up`
- For staging Cromwell: docker-compose -f cromwell-compose-staging.yml up

Note: this is a workaround until we implement the native timing diagram in Job Manager. Also, this workaround is mainly for Cromwell server.

### CODE REVIEW

**Comment by samanehsan** [comment]
Can you please look into the failing CI build @rexwangcc?

**Comment by rexwangcc** [comment]
Yeah, I'm looking into it, I suspect I need to update the UI test as well, sorry.

File: ui/src/app/job-details/tasks/tasks.component.html [hunk 1]
```diff
@@ -3,7 +3,14 @@
     <p class="title">Tasks</p>
     <md-tab-group>
       <md-tab label = "List view"></md-tab>
-      <md-tab label = "Graph view"></md-tab>
+      <!--<md-tab label = "Graph view"></md-tab>-->
+      <md-tab>
+        <ng-template md-tab-label>
+          <a style="all: unset" href="{{ serverUrl == '' ? '/jobs/' + parentJobId : serverUrl + parentJobId + '/timing' }}" target="_blank">
```
Comment (samanehsan) [inline ui/src/app/job-details/tasks/tasks.component.html]: I would move this logic into a function inside of the tasks component and call it in the template to clean this up a bit

File: ui/src/app/job-details/tasks/tasks.component.ts [hunk 2]
```diff
@@ -21,6 +22,10 @@ import {TaskMetadata} from '../../shared/model/TaskMetadata';
 })
 export class TaskDetailsComponent implements OnInit, OnChanges {
   @Input() tasks: TaskMetadata[] = [];
+  @Input() jobid: string;
```
Comment (bfcrampton) [inline ui/src/app/job-details/tasks/tasks.component.ts]: nit: camel case `jobId`

File: ui/src/app/job-details/tasks/tasks.component.ts [hunk 3]
```diff
@@ -35,10 +40,15 @@ export class TaskDetailsComponent implements OnInit, OnChanges {
 
   ngOnInit() {
     this.dataSource = new TasksDataSource(this.database);
+    this.parentJobId = this.jobid;
+    if (environment.serverUrl){
```
Comment (bfcrampton) [inline ui/src/app/job-details/tasks/tasks.component.ts]: nit: space between `)` and `{`

File: ui/src/environments/environment.cromwell.dev.ts [hunk 4]
```diff
@@ -5,5 +5,6 @@ export const environment = {
   requiresAuth: false,
   scope: '',
   additionalColumns: null,
-  entryPoint: 'jobs'
+  entryPoint: 'jobs',
+  serverUrl: 'https://cromwell.mint-dev.broadinstitute.org/api/workflows/v1/'
```
Comment (bfcrampton) [inline ui/src/environments/environment.cromwell.dev.ts]: nit: leave a trailing comma

File: ui/src/environments/environment.cromwell.staging.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const environment = {
+  apiUrl: '/api/v1',
+  clientId: '',
+  production: false,
+  requiresAuth: false,
+  scope: '',
+  additionalColumns: null,
+  entryPoint: 'jobs',
+  serverUrl: 'https://cromwell.mint-staging.broadinstitute.org/api/workflows/v1/'
```
Comment (bfcrampton) [inline ui/src/environments/environment.cromwell.staging.ts]: nit: trailing comma

File: ui/src/environments/environment.ts [hunk 6]
```diff
@@ -11,4 +11,5 @@ export const environment = {
   scope: '',
   additionalColumns: null,
   entryPoint: 'jobs',
+  serverUrl: ''
```
Comment (bfcrampton) [inline ui/src/environments/environment.ts]: You need to add this default value to both dsub environment files as well, right now running the google and local servers fails. Also trailing comma please.

File: ui/src/app/job-details/tasks/tasks.component.ts [hunk 7]
```diff
@@ -21,6 +22,10 @@ import {TaskMetadata} from '../../shared/model/TaskMetadata';
 })
 export class TaskDetailsComponent implements OnInit, OnChanges {
   @Input() tasks: TaskMetadata[] = [];
+  @Input() jobid: string;
+  parentJobId = '';
```
Comment (bfcrampton) [inline ui/src/app/job-details/tasks/tasks.component.ts]: I don't really understand why you need a `parentJobId` when its just a mirror of `jobId`. Aren't they set equal on init and whenever jobId changes? I might be missing something but it seems you could do with just `jobId`.

File: ui/src/app/job-details/tasks/tasks.component.ts [hunk 8]
```diff
@@ -35,10 +40,15 @@ export class TaskDetailsComponent implements OnInit, OnChanges {
...
     this.dataSource = new TasksDataSource(this.database);
+    this.parentJobId = this.jobid;
+    if (environment.serverUrl){
+      this.serverUrl = environment.serverUrl;
+    }
   }
 
   ngOnChanges(changes: SimpleChanges) {
     this.tasks = changes.tasks.currentValue;
+    this.parentJobId = changes.jobid.currentValue;
```
Comment (samanehsan) [inline ui/src/app/job-details/tasks/tasks.component.ts]: Is this line necessary? I wouldn't expect the `parentJobId` to change

File: ui/src/app/job-details/tasks/tasks.component.ts [hunk 9]
```diff
@@ -63,6 +61,14 @@ export class TaskDetailsComponent implements OnInit, OnChanges {
   getResourceFileName(url: string): string {
     return ResourceUtils.getResourceFileName(url);
   }
+
+  getTimingUrl(serverUrl: string, jobId: string): string {
+    if (serverUrl) {
+      return serverUrl + jobId + '/timing';
+    } else {
+      return '/jobs/' + jobId;
```
Comment (rexwangcc) [inline ui/src/app/job-details/tasks/tasks.component.ts]: Ideally this else should return something like `#` or `javascript:void(0);`, but for the former one from Angular2 changed the behavior from "don nothing" to "App root", and for latter one Angular automatically redirects it to `unsafe:javascript:void(0);` for security issues. Other alternatives require more work on module's configs, so I make it just refresh the current page if the `serverUrl` is '' (mostly for dsub)

File: ui/src/app/job-details/tasks/tasks.component.ts [hunk 10]
```diff
@@ -21,6 +22,9 @@ import {TaskMetadata} from '../../shared/model/TaskMetadata';
 })
 export class TaskDetailsComponent implements OnInit, OnChanges {
   @Input() tasks: TaskMetadata[] = [];
+  @Input() jobId: string = '';
+  serverUrl = '';
```
Comment (rexwangcc) [inline ui/src/app/job-details/tasks/tasks.component.ts]: This is a debug line, will defo remove it in next commit.

**Comment by rexwangcc** [comment]
The CI is passing so I'm going to merge this PR!

File: ui/src/app/job-details/tasks/tasks.component.ts [hunk 11]
```diff
@@ -40,15 +39,14 @@ export class TaskDetailsComponent implements OnInit, OnChanges {
 
   ngOnInit() {
     this.dataSource = new TasksDataSource(this.database);
-    this.parentJobId = this.jobid;
-    if (environment.serverUrl){
+    if (environment.serverUrl) {
       this.serverUrl = environment.serverUrl;
     }
+    this.serverUrl = '';
```
Comment (rexwangcc) [inline ui/src/app/job-details/tasks/tasks.component.ts]: This is a debug line, removed it in next commit.

**Pull request merged**